secant/Features/App/App.swift

```diff
@@ -43,6 +65,83 @@ extension AppReducer {
+                
+                switch (keysPresent, databaseFilesPresent) {
+                case (false, false):
+                    walletInitialisationState = .uninitialized
+                case (false, true):
+                    walletInitialisationState = .keysMissingFilesPresent
+                case (true, false):
+                    walletInitialisationState = .keysPresentFilesMissing
+                case (true, true):
+                    walletInitialisationState = .initialized
+                }
+            } catch CocoaError.fileNoSuchFile, CocoaError.fileReadNoSuchFile {
+                walletInitialisationState = keysPresent ? .keysPresentFilesMissing : .uninitialized
+            } catch {
+                walletInitialisationState = .failure(ErrorWrapper(error: error))
+            }
+            
+            return Effect(value: .walletInitialisationResult(walletInitialisationState))
+
+            /// All possible responses to the `.checkWalletInitialisation` action result.
+            /// `uninitialized` results in the start of the Onboarding flow for the user.
+        case .walletInitialisationResult(let walletInitialisationState):
+            switch walletInitialisationState {
+            case .uninitialized:
+                return Effect(value: .updateRoute(.onboarding))
+                    .delay(for: 3, scheduler: environment.mainQueue)
+                    .eraseToEffect()
+            case .initialized, .keysPresentFilesMissing:
+                return Effect(value: .initialiseApp)
+            case .keysMissingFilesPresent:
+                state.appInitialisationState = walletInitialisationState
+                // TODO: error we need to handle, issue #221 (https://github.com/zcash/secant-ios-wallet/issues/221)
+                return .none
+            case .failure(error: let error):
+                state.appInitialisationState = walletInitialisationState
+                // TODO: error we need to handle, issue #221 (https://github.com/zcash/secant-ios-wallet/issues/221)
+                return .none
+            }
+
+            /// Stored wallet is present, database files may or may not be present, trying to initialise app state variables and environments.
+            /// When initialisation succeeds user is taken to the home screen.
+        case .initialiseApp:
+            do {
+                state.storedWallet = try environment.recoveryPhraseStorage.exportWallet()
+            } catch {
+                state.appInitialisationState = .failure(ErrorWrapper(error: error))
+                // TODO: error we need to handle, issue #221 (https://github.com/zcash/secant-ios-wallet/issues/221)
+                return .none
+            }
+            
+            state.appInitialisationState = .initialized
+            return Effect(value: .updateRoute(.startup))
+                .delay(for: 3, scheduler: environment.mainQueue)
+                .eraseToEffect()
+
+        default:
```

Comment:
Do we need a default here or can we rely on the compiler to help us know when actions have been added?

Reply:
good catch, I don't think we need a default action

Reply:
the default here is a tradeoff. There's already AppReducer handling the routes and I made another AppReducer handling the actions. The routes one already had default. I think we are missing compilers switch check but I think this time it's worth it. Otherwise we would have plenty of unused cases. Thoughts @pacu @adamstener 

```
    private static let appReducer = AppReducer { state, action, environment in

    private static let routeReducer = AppReducer { state, action, _ in
```

Reply:
done